dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDLAPI.scala

```diff
@@ -0,0 +1,340 @@
+    Reshape(size.asScala.toArray)
+  }
+
+  //   Optimizer
+  def createClassNLLCriterion: ClassNLLCriterion[T] = {
+    ClassNLLCriterion[T]()
+  }
+
+  def createMSECriterion: MSECriterion[T] = {
+    MSECriterion[T]()
+  }
+
+  def createValidator(model: AbstractModule[Activity, Activity, T],
+                      valRDD: JavaRDD[PySample],
+                      batchSize: Int)
+  : Validator[T, MiniBatch[T]] = {
+    Validator(model, batching(valRDD, batchSize))
+  }
+
+  def test(validator: Validator[T, MiniBatch[T]], valMethods: JList[String])
+  : JList[TestResult] = {
+    val resultArray = validator.test(toValidationMethod(valMethods))
+    val testResultArray = resultArray.map{result =>
+      TestResult(result._1.result()._1, result._1.result()._2,
+        validationMethodToStr(result._2))
+    }
+    println(s"inference result len: ${testResultArray.length}")
+    testResultArray.foreach(println(_))
+    testResultArray.toList.asJava
+  }
+
+  def modelFromPath(path: String): AbstractModule[Activity, Activity, T] = {
+    Module.load[T](path)
+  }
+
+  def modelPredictRDD(model: AbstractModule[Activity, Activity, T],
+                      dataRdd: JavaRDD[PySample]): JavaRDD[PySample] = {
+    val result = predict(model, dataRdd.rdd.map(toSample(_)))
+    result.map(toPySample(_))
+
+  }
+
+  def modelGetParameters(model: AbstractModule[Activity, Activity, T])
+  : PySample = {
+    model.getParameters()
+    val result = new JArrayList[JArrayList[T]]()
+    val item = new JArrayList[JList[T]]()
+
+    val weightTensor = model.getParameters()._1.contiguous()
+    val weightsData: JList[T] = weightTensor.storage().toList.asJava
+    val weightsShape = weightTensor.size().toList.asJava
+
+    val gradientTensor = model.getParameters()._1.contiguous()
+    val gradientData: JList[T] = gradientTensor.storage().toList.asJava
+    val gradientShape = gradientTensor.size().toList.asJava
+    // TODO: remove float
+    PySample(weightsData.asInstanceOf[JList[Any]],
+      gradientData.asInstanceOf[JList[Any]], weightsShape, gradientShape, "float")
+  }
+
+  def predict(model: AbstractModule[Activity, Activity, T],
+                      dataRdd: RDD[Sample[T]]): RDD[Sample[T]] = {
+    val modelBroadCast = dataRdd.sparkContext.broadcast(model.evaluate())
+    dataRdd.mapPartitions {partition =>
+      val localModel = modelBroadCast.value.cloneModule()
+      partition.map {sample =>
+        val output = localModel.forward(sample.feature()).toTensor[T]
+        Sample(sample.feature(), output)
+      }
+    }
+  }
+
+  def createMaxEpoch(max: Int): Trigger = {
+    Trigger.maxEpoch(max)
+  }
+
+  def createEveryEpoch(): Trigger = {
+    Trigger.everyEpoch
+  }
+
+  def createSeveralIteration(interval: Int): Trigger = {
+    Trigger.severalIteration(interval)
+  }
+
+  def createMaxIteration(max: Int): Trigger = {
+    Trigger.maxIteration(max)
+  }
+
+  def createOptimizer(model: AbstractModule[Activity, Activity, T],
```

Comment:
We should add an Optimizer.apply() method with the same signature, and then have this method to call that.